retail/interactive-tutorial/RetailProducts.Samples/UpdateProductSample.cs

```diff
@@ -0,0 +1,146 @@
+    /// <param name="length">The required length of alphanumeric string.</param>
+    /// <returns>Generated alphanumeric string.</returns>
+    private static string RandomAlphanumericString(int length)
+    {
+        const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+        return new string(Enumerable.Repeat(chars, length)
+            .Select(s => s[Random.Next(s.Length)]).ToArray());
+    }
+
+    /// <summary>
+    /// Generate product for update.
+    /// </summary>
+    /// <param name="productId">The actual product id.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>Generated retail product.</returns>
+    private static Product GenerateProductForUpdate(string productId, string projectId)
+    {
+        var updatedPriceInfo = new PriceInfo
+        {
+            Price = 20.0f,
+            OriginalPrice = 25.5f,
+            CurrencyCode = "EUR"
+        };
+
+        var generatedProduct = new Product
+        {
+            Id = productId,
+            Name = $"projects/{projectId}/locations/global/catalogs/default_catalog/branches/default_branch/products/{productId}",
+            Title = "Updated Nest Mini",
+            Type = Product.Types.Type.Primary,
+            PriceInfo = updatedPriceInfo,
+            Availability = Product.Types.Availability.OutOfStock
+        };
+
+        generatedProduct.Categories.Add("Updated Speakers and displays");
+        generatedProduct.Brands.Add("Updated Google");
+
+        return generatedProduct;
+    }
+
+    /// <summary>
+    /// Get the update product request.
+    /// </summary>
+    /// <param name="productToUpdate">The product to update object.</param>
+    /// <returns>Update product request.</returns>
+    private static UpdateProductRequest GetUpdateProductRequest(Product productToUpdate)
+    {
+        var updateProductRequest = new UpdateProductRequest
+        {
+            Product = productToUpdate,
+            AllowMissing = true
+        };
+
+        Console.WriteLine("Update product. request:");
+        Console.WriteLine(updateProductRequest);
+        Console.WriteLine();
+
+        return updateProductRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to update a product.
+    /// </summary>
+    /// <param name="originalProduct">The original product object.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>Updated retail product.</returns>
+    public static Product UpdateRetailProduct(Product originalProduct, string projectId)
+    {
+        var productForUpdate = GenerateProductForUpdate(originalProduct.Id, projectId);
+        var updateProductRequest = GetUpdateProductRequest(productForUpdate);
+        ProductServiceClient client = ProductServiceClient.Create();
+        Product updatedProduct = client.UpdateProduct(updateProductRequest);
+
+        Console.WriteLine("Updated product:");
+        Console.WriteLine(updatedProduct);
```

Comment:
Consider being more explicit:
```suggestion
        Console.WriteLine($"Update title: {updatedProduct.Title}");
        ...
        ...
```

Reply:
Sure. Improved the console outputs in all samples.